delphi-frontend/src/main/java/au/com/integradev/delphi/operator/OperatorInvocableCollector.java

```diff
@@ -270,58 +320,129 @@ private Set<Invocable> createAnd() {
+    if (!result.isEmpty()) {
+      return result;
+    }
 
     Type int8 = typeFactory.getIntrinsic(IntrinsicType.SHORTINT);
     Type int16 = typeFactory.getIntrinsic(IntrinsicType.SMALLINT);
     Type int32 = typeFactory.getIntrinsic(IntrinsicType.INTEGER);
     Type int64 = typeFactory.getIntrinsic(IntrinsicType.INT64);
-    result.addAll(createWithInterleavedTypes(bitwiseName, int8, int16, int32, int64));
 
     Type uint8 = typeFactory.getIntrinsic(IntrinsicType.BYTE);
     Type uint16 = typeFactory.getIntrinsic(IntrinsicType.WORD);
     Type uint32 = typeFactory.getIntrinsic(IntrinsicType.CARDINAL);
     Type uint64 = typeFactory.getIntrinsic(IntrinsicType.UINT64);
-    result.addAll(createWithInterleavedTypes(bitwiseName, uint8, uint16, uint32, uint64));
+
+    Type widenedUInt16 = ((TypeFactoryImpl) typeFactory).widenedUInt16();
+    Type widenedUInt32 = ((TypeFactoryImpl) typeFactory).widenedUInt32();
+
+    addAll(
+        result,
+        new OperatorIntrinsic(name, List.of(int8, int8), int8),
+        new OperatorIntrinsic(name, List.of(int16, int16), int16),
+        new OperatorIntrinsic(name, List.of(uint8, uint8), uint8),
+        new OperatorIntrinsic(name, List.of(widenedUInt16, widenedUInt16), widenedUInt16),
+        new OperatorIntrinsic(name, List.of(uint16, uint16), uint16),
+        new OperatorIntrinsic(name, List.of(int32, int32), int32),
+        new OperatorIntrinsic(name, List.of(int32, int64), int64),
+        new OperatorIntrinsic(name, List.of(int64, int32), int64),
+        new OperatorIntrinsic(name, List.of(widenedUInt32, widenedUInt32), widenedUInt32),
+        new OperatorIntrinsic(name, List.of(uint32, widenedUInt32), uint32),
+        new OperatorIntrinsic(name, List.of(widenedUInt32, uint32), uint32),
+        new OperatorIntrinsic(name, List.of(uint32, uint32), uint32),
+        new OperatorIntrinsic(name, List.of(int64, int64), int64),
+        new OperatorIntrinsic(name, List.of(int64, uint64), uint64),
+        new OperatorIntrinsic(name, List.of(uint64, int64), uint64),
+        new OperatorIntrinsic(name, List.of(uint64, uint64), uint64));
 
     return result;
   }
 
-  private Set<Invocable> createArithmeticBinary(String name) {
-    Type integer = typeFactory.getIntrinsic(IntrinsicType.INTEGER);
-    Type extended = typeFactory.getIntrinsic(IntrinsicType.EXTENDED);
+  private Set<Invocable> createNativeIntegerBinary(String name) {
+    Set<Invocable> result = Sets.newHashSet();
 
-    return addAll(
-        createIntegerArithmeticBinary(name),
-        new OperatorIntrinsic(name, List.of(extended, extended), extended),
-        new OperatorIntrinsic(name, List.of(integer, extended), extended),
-        new OperatorIntrinsic(name, List.of(extended, integer), extended));
+    Type nativeInt = typeFactory.getIntrinsic(IntrinsicType.NATIVEINT);
+    Type nativeUInt = typeFactory.getIntrinsic(IntrinsicType.NATIVEUINT);
+
+    if (!nativeInt.isWeakAlias()) {
```

Comment:
This could be `nativeInt.isStrongAlias()`. Did you choose to negate the `weak` version because that was the terminology used in the Delphi 12 release?

Reply:
It's actually because SonarDelphi doesn't model the unique `Native*` types as strong aliases at all - but rather as distinct integer types with the same value range as the platform-specific integer types.

Reply:
Right, I was thinking of `isStrongAlias` and `isWeakAlias` and being the negative of one another because I was just looking at the impl in `AliasTypeImpl`. The types could of course not be aliases at all.

I guess this is dead code until we do #81.